app/Console/Command/OwnershipCommand.php

```diff
@@ -0,0 +1,155 @@
+            ->setDescription('Add ownership via an UiTID or email')
+            ->addArgument(
+                self::ITEM_TYPE,
+                InputOption::VALUE_REQUIRED,
+                'The type(at present only organizer is allowed) of the item to which you want to add the ownership.'
+            )
+            ->addArgument(
+                self::ITEM_ID,
+                InputOption::VALUE_REQUIRED,
+                'The id of the item to which you want to add the ownership.'
+            )
+            ->addArgument(
+                self::USER,
+                InputOption::VALUE_REQUIRED,
+                'The id or email of a user give ownership to.'
+            );
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $itemType = $input->getArgument(self::ITEM_TYPE);
+        $itemId = $input->getArgument(self::ITEM_ID);
+        $user = $input->getArgument(self::USER);
+
+        if (!$this->itemExists($itemId)) {
+            $output->writeln('Organizer does not exist.');
+            return self::FAILURE;
+        }
+
+        $userId = $this->getUserId($user);
+        if ($userId === null) {
+            $output->writeln('No user found for ' . $user);
+            return self::FAILURE;
+        }
+
+        if ($this->ownershipExist($itemId, $userId)) {
+            $output->writeln('Ownership already exists.');
+            return self::FAILURE;
+        }
+
+        $ownerShipId = new Uuid($this->uuidFactory->uuid4()->toString());
+
+        $requestOwnership = new RequestOwnership(
+            $ownerShipId,
+            new Uuid($itemId),
+            new ItemType($itemType),
+            $userId,
+            new UserId(UUID::NIL)
+        );
+
+        $this->commandBus->dispatch($requestOwnership);
+
+        $approveOwnership = new ApproveOwnership($ownerShipId);
+
+        $this->commandBus->dispatch($approveOwnership);
+
+        return self::SUCCESS;
+    }
+
+    private function ownershipExist(string $itemId, UserId $ownerId): bool
+    {
+        $existingOwnershipItems = $this->ownershipSearchRepository->search(
+            new SearchQuery([
+                new SearchParameter('itemId', $itemId),
+                new SearchParameter('ownerId', $ownerId->toString()),
+                new SearchParameter('state', OwnershipState::requested()->toString()),
+                new SearchParameter('state', OwnershipState::approved()->toString()),
+            ])
+        );
+        return $existingOwnershipItems->count() > 0;
+    }
+
+    private function getUserId(string $user): ?UserId
+    {
+        try {
+            $userEmail = new EmailAddress($user);
+            $user = $this->userIdentityResolver->getUserByEmail($userEmail);
+            if (!$user) {
+                return null;
+            }
+            return new UserId($user->getUserId());
+        } catch (InvalidEmailAddress $exception) {
+            return new UserId($user);
```

Comment:
This fallback can be tricky. For example, if someone made a mistake and provided `john.doe@gmail` as a parameter. This will result in an `InvalidEmailAddress` exception, and the value will be cast to `UserId. This will work because it is a string and not empty. Maybe we also need to call `getUserById` to make sure the user is found?

I think the HTTP request handler suffers from the same issue, there is no validation to match a passed user ID to an existing user. So the HTTP layer accepts random user IDs. Maybe we need an extra acceptance test for this?

Reply:
Good point, I've fixed this for the CLI Command & made a ticket for the HTTP endpoint => https://jira.publiq.be/browse/III-6622